### PR TITLE
Add type options to title popover in FilterList

### PR DESCRIPTION
**Pull request by michsa**
Closes an issue 🎉

Branched from #250, and depends on it for `newNodeFromType` (I think that's actually the only thing, though...)

### CODE REVIEW

**Comment by decrapifier** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>The README has not been updated. Please update the README.</td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>Could not find any browser results.</td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>We were able to automatically fix some formatting issues in this PR for you!</td>
    </tr>
  </tbody>
</table>


### Some things that were possibly fixed:

- Code that could be fixed via the --fix flag
- Formatting that could be fixed by prettier

Take a look at this commit to see what happened in detail: 3f40f0eca3b23a0dcaabb86c2cbb0f960e01fb88


And look at this wiki page to see the reasoning behind the ESLint rules: https://github.com/smartprocure/eslint-config-smartprocure/wiki/Rules-and-Why-We-Chose-Them
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 0e5e18221e85f05e789abea26fb09c63e4538a97
</p>

File: src/FilterList.js [hunk 1]
```diff
@@ -39,11 +40,43 @@ export let Label = inject(_.pick('tree'))(
...
+                  ),
+                  F.autoLabelOptions(
+                    _.without(
+                      [node.type],
+                      _.get([node.field, 'typeOptions'], fields)
+                    ) || []
+                  )
+                )}
+                <div
+                  style={{ borderBottom: '1px solid #eee', margin: '4px -5px' }}
```
Comment (daedalus28) [inline src/FilterList.js]: This probably shouldn't be in the core components and should be a themeable thing, e.g. something like `className="dropdown-option-separator"` or whatever and then themed in grey vest

**Comment by stellarhoof** [comment]
@michsa Can you add a screenshot?

**Comment by michsa** [comment]
@stellarhoof here is it in grey vest:

Screen Shot 2019-06-26 at 11 32 56 AM

And in blueberry (unstyled):

Screen Shot 2019-06-26 at 11 33 39 AM

**Comment by stellarhoof** [comment]
I'm not sure what the policy on new files is, but my first instinct would be to keep `FilterOptions` inside `FilterList`, unless you've got plans for it somewhere else.

Also what do you think about `FilterActions`?

Otherwise looks good.

Oh PS: Why the dash at the beginning of the filter types?

**Comment by michsa** [comment]
Yeah, `filterActions` might be better. And you're probably right about just keeping it inside `FilterList` - we can always factor it out later if necessary. Plus, this is actually only half the feature - there will be also an option to change the node's field, which should open a modal picker. Making that work from inside a popover is pretty problematic, but if it's all inside `FilterList` hopefully it will be easier.

And the dash wasn't my idea (but I do think it looks nice)

File: src/FilterOptions.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import React from 'react'
+import _ from 'lodash/fp'
+import F from 'futil-js'
+import { newNodeFromType } from './utils/search'
+
+let FilterOptions = ({ node, tree, fields, Item }) => (
```
Comment (daedalus28) [inline src/FilterOptions.js]: agreed with @stellarhoof - this doesn't mean much outside of `FilterList`, and `FilterActions` is probably a better name

File: src/FilterOptions.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import React from 'react'
```
Comment (stellarhoof) [inline src/FilterOptions.js]: You forgot to remove this file :D

Comment (michsa) [reply src/FilterOptions.js]: 🤦‍♀

File: src/themes/greyVest/index.js [hunk 4]
```diff
@@ -490,6 +490,23 @@ export let GVStyle = () => (
         padding-top: 30px;
       }
 
+      .filter-options-popover {
```
Comment (stellarhoof) [inline src/themes/greyVest/index.js]: It'd be nice if these were `filter-actions-*` as well

**Pull request merged**